rest_api/index.adoc

```diff
@@ -7,8 +7,171 @@
 :toc: macro
 :toc-title:
 
+toc::[]
+
 The OpenShift distribution of Kubernetes includes the link:./kubernetes_v1.html[Kubernetes v1 REST API]
 and the link:./openshift_v1.html[OpenShift v1 REST API]. These are RESTful APIs accessible via HTTP(s)
 on the OpenShift master servers.
 
-These REST APIs can be used to manage end-user applications, the cluster, and the users of the cluster.
\ No newline at end of file
+These REST APIs can be used to manage end-user applications, the cluster, and the users of the cluster.
+
+== Authentication
+
+API calls must be authenticated with an access token or X.509 certificate. See
+link:../architecture/additional_concepts/authentication.html#api-authentication[authentication architecture]
+for an overview.
+
+This section highlights the token authentication method. With token
+authentication a bearer token must be passed in as an
+link:https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.8[HTTP Authorization header].
+There are two types of access tokens, session and service account.
+
+A **session** token is short-lived, expiring within 24 hours by default. It
+represents a user. The session token may be obtained with this command:
+
+----
+$ oc whoami -t
+dIAo76N-W-GXK3S_w_KsC6DmH3MzP79zq7jbMQvCOUo
+----
+
+**Service account** tokens are long-lived tokens. They are
+link:https://tools.ietf.org/html/rfc7519[JSON Web Token (JWT)] formatted tokens
+and are much longer strings than session tokens. See the
+link:../dev_guide/service_accounts.html#using-a-service-account-s-credentials-externally[Developer Guide]
+for more information.
+
+A service account token may be obtained with these commands:
+
+[options="nowrap"]
+----
+$ oc create serviceaccount robot
+serviceaccount "robot" created
+
+$ oc describe serviceaccount robot
+Name:		robot
+Namespace:	test
+Labels:		<none>
+
+Image pull secrets:	robot-dockercfg-rdrpg
+
+Mountable secrets: 	robot-token-2dsne
+                   	robot-dockercfg-rdrpg
+
+Tokens:            	robot-token-2dsne
+                   	robot-token-9efwm
+
+$  oc describe secret robot-token-2dsne
```

Comment:
I agree 100% except I ran into this: https://bugzilla.redhat.com/show_bug.cgi?id=1333560